crypto/src/main/java/org/cryptimeleon/incentive/crypto/proof/spend/leaf/MetadataLeaf.java

```diff
@@ -0,0 +1,17 @@
+package org.cryptimeleon.incentive.crypto.proof.spend.leaf;
+
+import org.cryptimeleon.incentive.crypto.proof.spend.tree.SpendDeductLeafNode;
+import org.cryptimeleon.incentive.crypto.proof.spend.zkp.SpendDeductBooleanZkp;
+
+/**
+ * Leaf node for a ZKP that checks the metadata of the tokens for which the user must always know a witness.
+ * Since it does not make sense to have two of these in the same {@link SpendDeductBooleanZkp},
+ * the name constant.
+ */
+public class MetadataLeaf extends SpendDeductLeafNode {
```

Comment:
I don't quite understand what function this serves. This contains no data. And I thought the `SpendDeductLeafNode`s are (API-)user-facing. I wouldn't let the user decide whether or not to include this (also wouldn't bother them with it, from a usability perspective).

Instead, adapt the transformation to just always add the corresponding Sigma protocol to the tree.

Reply:
I hardcoded the AND with the MetadataZkp in the subclass of ProofOfPartialKnowledge.